UI/window-basic-settings-stream.cpp

```diff
@@ -208,14 +224,21 @@ void OBSBasicSettings::LoadStream1Settings()
 void OBSBasicSettings::SaveStream1Settings()
 {
 	bool customServer = IsCustomService();
-	const char *service_id = customServer ? "rtmp_custom" : "rtmp_common";
+	bool whip = IsWHIP();
+	const char *service_id = "rtmp_common";
+
+	if (customServer) {
+		service_id = "rtmp_custom";
+	} else if (whip) {
+		service_id = "whip_custom";
+	}
```

Comment:
this appears to be duplicated in another place (`UI/window-basic-settings-stream.cpp`, `SpawnTempService`), could it be factored out?